src/mokernels/intrinsiccoregion.jl

```diff
@@ -34,10 +34,36 @@ function IntrinsicCoregionMOKernel(; kernel::Kernel, B::AbstractMatrix)
     return IntrinsicCoregionMOKernel{typeof(kernel),typeof(B)}(kernel, B)
 end
 
+function IntrinsicCoregionMOKernel(kernel::Kernel, B::AbstractMatrix)
+    return IntrinsicCoregionMOKernel{typeof(kernel),typeof(B)}(kernel, B)
+end
+
 function (k::IntrinsicCoregionMOKernel)((x, px)::Tuple{Any,Int}, (y, py)::Tuple{Any,Int})
     return k.B[px, py] * k.kernel(x, y)
 end
 
+function matrixkernel(k::IntrinsicCoregionMOKernel, x, y)
+    return matrixkernel(k, x, y; outputsize=size(k.B, 1))
```

Comment:
minor suggestion:
```suggestion
    return matrixkernel(k, x, y; outputsize=size(k.B, 2))
```
doesn't make any difference as `B` must be square, but then it's more consistent with the LMC one:)